coffee-chats/src/main/java/com/google/step/coffee/data/RequestStore.java

```diff
@@ -3,26 +3,89 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.users.User;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.step.coffee.HttpError;
 import com.google.step.coffee.entity.ChatRequest;
+import com.google.step.coffee.entity.ChatRequestBuilder;
+import com.google.step.coffee.entity.TimeSlot;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
 
 public class RequestStore {
   private DatastoreService datastore;
+  private TagStore tagStore;
+
 
   public RequestStore() {
     this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.tagStore = new TagStore();
   }
 
-  public void addRequest(ChatRequest request, String userId) {
+  public void addRequest(ChatRequest request) {
     Entity reqEntity = new Entity("ChatRequest");
     reqEntity.setProperty("tags", request.getTags());
     reqEntity.setProperty("dates", request.getDates());
     reqEntity.setProperty("minPeople", request.getMinPeople());
     reqEntity.setProperty("maxPeople", request.getMaxPeople());
-    reqEntity.setProperty("randomMatch", request.shouldMatchRandom());
-    reqEntity.setProperty("matchRecent", request.shouldMatchRecents());
-    reqEntity.setProperty("userId", userId);
+    reqEntity.setProperty("duration", request.getDuration());
+    reqEntity.setProperty("matchRandom", request.shouldMatchRandom());
+    reqEntity.setProperty("matchRecents", request.shouldMatchRecents());
+    reqEntity.setProperty("userId", request.getUserId());
 
     datastore.put(reqEntity);
+    tagStore.addTags(request.getTags());
+  }
+
+  public List<ChatRequest> getUnmatchedRequests() throws HttpError {
```

Comment:
I don't think it should be throwing `HttpError`, probably is best to create some new exception types, since I don't see a reason why the requests need to know about the communication protocol between our server and clients.

Probably `ChatRequestBuilder` should not throw `HttpError`, but some other exception type.

Reply:
That seems more sensible. Done